File: WaterManager/src/main/java/pissir/watermanager/model/item/RichiestaIdrica.java
package pissir.watermanager.model.item;

import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

/**
 * @author Almasio Luca
 * @author Borova Dritan
 * @author Gattico Alessandro
 */
@Setter
@Getter
@NoArgsConstructor
public class RichiestaIdrica {
	
	private int id;
	private Double quantita;
	private int idColtivazione;
	private int idBacino;
	private String date;
	private Approvazione approvato;
	private String nomeAzienda;
	
	
	public RichiestaIdrica (int id, Double quantita, int idColtivazione, int idBacino, String date, String nomeAzienda) {
		this.id = id;
		this.quantita = quantita;
		this.idColtivazione = idColtivazione;
		this.idBacino = idBacino;
		this.date = date;
		this.nomeAzienda = nomeAzienda;
	}
	
}

File: WaterManagerUI/Model/Item/RichiestaIdrica.cs
using System.Runtime.InteropServices.JavaScript;
using Newtonsoft.Json;

namespace WaterManagerUI.Model.Item;

public class RichiestaIdrica
{
    public int id { get; set; }
    public Double quantita { get; set; }
    public int idColtivazione { get; set; }
    public int idBacino { get; set; }
    public String date { get; set; }
    public Approvazione approvato { get; set; }
    public String nomeAzienda { get; set; }

    public RichiestaIdrica()
    {
    }

    public RichiestaIdrica(Double quantita, int idColtivazione, int idBacino, String date, string nomeAzienda)
    {
        this.id = 0;
        this.quantita = quantita;
        this.idColtivazione = idColtivazione;
        this.idBacino = idBacino;
        this.date = date;
        this.nomeAzienda = nomeAzienda;
    }

    [JsonConstructor]
    public RichiestaIdrica(int id, double quantita, int idColtivazione, int idBacino, string date, string nomeAzienda,
        Approvazione approvato)
    {
        this.id = id;
        this.quantita = quantita;
        this.idColtivazione = idColtivazione;
        this.idBacino = idBacino;
        this.date = date;
        this.nomeAzienda = nomeAzienda;
        this.approvato = approvato;
    }


    public DateTime RichiestaAsDateTime()
    {
        DateTime date;
        if (DateTime.TryParse(this.date, out date))
        {
            return date;
        }
        else
        {
            return DateTime.MinValue;
        }
    }
}